src/cpu/reorder/simple_reorder.hpp

```diff
@@ -355,21 +361,32 @@ struct simple_reorder_impl<SIMPLE_REORDER_TEMPL_CALL,
 
         const auto &plain_d = order_keep ? input_d : output_d;
         const auto &dims = input_d.dims();
+        const int ndims = input_d.ndims();
         const auto &pdims
                 = order_keep ? output_d.padded_dims() : input_d.padded_dims();
 
         const dim_t G = w_groups ? dims[0] : 1;
         const dim_t OC = dims[w_groups + 0];
+        const dim_t PADDED_OC = pdims[w_groups + 0];
         const dim_t NB_OC = pdims[w_groups + 0] / ocblksize;
         const dim_t IC = dims[w_groups + 1];
         const dim_t NB_IC = pdims[w_groups + 1] / icblksize;
         const dim_t D = is_3d ? dims[2 + w_groups] : 1;
         const dim_t H = is_1d || is_0d ? 1 : dims[2 + w_groups + is_3d];
         const dim_t W = is_0d ? 1 : dims[w_groups + is_3d + 3 - is_1d];
 
+        int smask = pd->attr()->output_scales_.mask_;
+        // XXX: Currently user can pass a mask that has non-zero values in
+        // dimensions that do not exist in a md. Since attributes are created
+        // separately mask can't be validated.
+        // This line truncates a given mask in range [0, 1 << ndims - 1]
+        // TODO: Such masks can be either prohibited at pd creation step at
+        // API level or checked by each implementation that relies on it.
+        smask &= (1 << ndims) - 1;
```

Comment:
Considering more templated branches may appear, I vote for restricting masks and implement a proper mask passing from benchdnn to be safer in future.

Reply:
Discussed offline. API-level changes will require a lot of work, while only couple of implementations are affected by the current behavior. Desicion is to keep it as is.